streamlit_app.py

```diff
@@ -42,13 +54,18 @@ def get_word_df(words):
 
 
 def draw_tweet_data(stopwords, representation, state=None):
-    cleaned_tweets = get_cleaned_tweet_words(state, stopwords)
     if representation == "Word Cloud":
-        wordcloud = get_wordcloud(cleaned_tweets, state)
-        st.pyplot(wordcloud)
+        cached_pic = get_wordcloud_from_file(state)
+        if cached_pic:
+            st.image(cached_pic, width=600)
+        else:
+            cleaned_tweets = get_cleaned_tweet_words(state, stopwords)
+            wordcloud = get_wordcloud(cleaned_tweets, state)
+            st.pyplot(wordcloud)
     else:
+        cleaned_tweets = get_cleaned_tweet_words(state, stopwords)
         title = f"{state} Tweets" if state else "Global Tweets"
```

Comment:
Not sure about this question but wanted to double-check, are these really global tweets or just US tweets?

Reply:
The site doesn't actually specify where the tweets are from but they never mention limiting them to one country: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/LW0BTB&version=7.0. Also when I was fetching and filtering them I was seeing tweets from other countries.

Reply:
Ah cool, thanks for the clarification!